### PR TITLE
Save endpoint and dirty state propagation

### LINKED ISSUE #78: Dirty state should be propagated

### LINKED ISSUE #79: Provide Save command

### PR DESCRIPTION
**Pull request by ndoschek**
- Add endpoint to save a resource with a certain modeluri
- Slim JsonResponse to two props type and data
- Introduce new JsonResponse types
- Minor changes of JsonResponse consumers

Fixes #79 
Fixes #78

### CODE REVIEW

File: com.eclipsesource.modelserver.client/src/test/java/com/eclipsesource/modelserver/client/ModelServerClientTest.java [hunk 1]
```diff
@@ -276,6 +276,6 @@ private ModelServerClient createClient() throws MalformedURLException {
     }
 
     private ObjectNode createDataResponse(JsonNode payload) {
-        return Json.object(Json.prop("data", payload));
+        return (ObjectNode) Json.merge(Json.object(Json.prop("type", Json.text("success"))), Json.object(Json.prop("data", payload)));
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.client/src/test/java/com/eclipsesource/modelserver/client/ModelServerClientTest.java]: Why not use `JsonResponse` here?

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/JsonResponse.java [hunk 2]
```diff
@@ -21,41 +21,69 @@
...
+ *   "data": String | Boolean | EObject
+ * }
+ */
 
-	public static ObjectNode success() {
+public class JsonResponse {
+	
+	private static ObjectNode type(String type) {
+		// Allowed types: success, error, fullUpdate, incrementalUpdate
+		return Json.object(Json.prop("type", Json.text(type)));
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/JsonResponse.java:37]: We should think about introducing an enum for the different `type`s. If you agree, let's create a followup issue.

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelController.java [hunk 3]
```diff
@@ -194,7 +204,7 @@ public void executeCommand(Context ctx, String modelURI) {
 											resource.get().getResourceSet().getResources().remove(resource.get());
 										}
 									});
-							ctx.json(JsonResponse.confirm("Model '" + modelURI + "' successfully updated"));
+							// ctx.json(JsonResponse.success("Model '" + modelURI + "' successfully updated"));
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelController.java]: Remove comment

**Review by edgarmueller — APPROVED** [review]

File: com.eclipsesource.modelserver.client/src/test/java/com/eclipsesource/modelserver/client/ModelServerClientTest.java (hunk 1 above)
Comment (ndoschek) [reply com.eclipsesource.modelserver.client/src/test/java/com/eclipsesource/modelserver/client/ModelServerClientTest.java]: Good point, will change that

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/JsonResponse.java (hunk 2 above)
Comment (ndoschek) [reply com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/JsonResponse.java:37]: Yes I agree, will create an issue for that

Comment (ndoschek) [reply com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/JsonResponse.java:37]: Issue #107

**Pull request merged**